components/org.wso2.carbon.identity.provisioning.connector.google/src/test/java/org/wso2/carbon/identity/provisioning/connector/google/GoogleProvisioningConnectorConfigTest.java

```diff
@@ -0,0 +1,137 @@
+    private static final int CASE_2 = 2;
+    private static final int CASE_3 = 3;
+
+    @Mock
+    private Log log;
+
+    @DataProvider(name = "requiredFieldsProvider")
+    public Object[][] provideTestData() {
+
+        Properties properties1 = new Properties();
+        properties1.setProperty(GoogleConnectorConstants.PropertyConfig.REQUIRED_FIELDS, REQUIRED_FIELDS);
+
+        Properties properties2 = new Properties();
+
+        return new Object[][]{
+                {properties1, false},
+                {properties2, true}
+        };
+    }
+
+    @Test(dataProvider = "requiredFieldsProvider")
+    public void testGetRequiredAttributeNames(Properties properties, boolean isEmpty) throws Exception {
+
+        setLogging(false);
+        GoogleProvisioningConnectorConfig connectorConfig = new GoogleProvisioningConnectorConfig(properties);
+
+        if (isEmpty) {
+            Assert.assertEquals(connectorConfig.getRequiredAttributeNames().size(), 0);
+        } else {
+            Assert.assertNotEquals(connectorConfig.getRequiredAttributeNames().size(), 0);
+        }
+    }
+
+    @Test(dataProvider = "requiredFieldsProvider")
+    public void testGetValue(Properties properties, boolean isEmpty) throws Exception {
+
+        setLogging(false);
+        GoogleProvisioningConnectorConfig connectorConfig = new GoogleProvisioningConnectorConfig(properties);
+
+        if (isEmpty) {
+            Assert.assertNull(connectorConfig.getValue(GoogleConnectorConstants.PropertyConfig.REQUIRED_FIELDS));
+        } else {
+            Assert.assertEquals(connectorConfig.getValue(GoogleConnectorConstants.PropertyConfig.REQUIRED_FIELDS),
+                                REQUIRED_FIELDS);
+        }
+    }
+
+    @DataProvider(name = "userIdProvider")
+    public Object[][] provideUserIdData() {
+
+        Properties properties1 = new Properties();
+        properties1.setProperty(GoogleConnectorConstants.PropertyConfig.USER_ID_CLAIM, LAST_NAME_CLAIM);
+        Properties properties2 = new Properties();
+        properties2.setProperty(GoogleConnectorConstants.ATTRIBUTE_PRIMARYEMAIL, EMAIL_NAME_CLAIM);
+        Properties properties3 = new Properties();
+
+        return new Object[][]{
+                // CASE_1: USER_ID_CLAIM property is present in connector configs.
+                {properties1, false, CASE_1},
+                // CASE_2: ATTRIBUTE_PRIMARYEMAIL property is present in connector configs.
+                {properties2, true, CASE_2},
+                // CASE_3: No external properties in connector configs. Default values are set.
+                {properties3, false, CASE_3},
+        };
+    }
+
+    @Test(dataProvider = "userIdProvider")
+    public void testGetUserIdClaim(Properties properties, boolean debugEnabled, int caseNo) throws Exception {
+        setLogging(debugEnabled);
+        GoogleProvisioningConnectorConfig connectorConfig = new GoogleProvisioningConnectorConfig(properties);
+
+        if (caseNo == CASE_1) {
```

Comment:
We can pass the expected value from  DataProvider  and reduce three asserts to a single one.